superset-frontend/src/SqlLab/components/QueryTable.jsx

```diff
@@ -74,151 +75,162 @@ class QueryTable extends React.PureComponent {
   }
 
   render() {
-    const data = this.props.queries
-      .map(query => {
-        const q = { ...query };
-        if (q.endDttm) {
-          q.duration = fDuration(q.startDttm, q.endDttm);
-        }
-        const time = moment(q.startDttm).format().split('T');
-        q.time = (
-          <div>
-            <span>
-              {time[0]} <br /> {time[1]}
-            </span>
-          </div>
-        );
-        q.user = (
-          <Button
-            buttonSize="small"
-            buttonStyle="link"
-            onClick={this.props.onUserClicked.bind(this, q.userId)}
-          >
-            {q.user}
-          </Button>
-        );
-        q.db = (
-          <Button
-            buttonSize="small"
-            buttonStyle="link"
-            onClick={this.props.onDbClicked.bind(this, q.dbId)}
-          >
-            {q.db}
-          </Button>
-        );
-        q.started = moment(q.startDttm).format('HH:mm:ss');
-        q.querylink = (
-          <div style={{ width: '100px' }}>
+    const data = memoize(() =>
```

Comment:
Mentioned a concern in another PR about the risks of `lodash/memoize` but long story short, it's got issues, and we should _probably_ use `memoize-one` for memoizing in class components OR conver them to function components and go with `useMemo`

Reply:
I refactored it into functional component